### PR TITLE
Update IDAES requirement to 2.2.0.dev0.watertap.23.08.03 and other stability improvements

### LINKED ISSUE #929: Make WaterTAP compatible with NL_v2 writer
We should remove the kludge introduced by #928 for Pyomo 6.5+.

See the test failures here: https://github.com/watertap-org/watertap/actions/runs/4198316702/jobs/7281778885

### LINKED ISSUE #1072: Fix OS-specific solver issues for NF flowsheet
See e.g.:

- [x] https://github.com/watertap-org/watertap/pull/1034#issuecomment-1614665539
- [x] #1062
- [ ] #1076

### PR DESCRIPTION
**Pull request by bknueven**
## Changes proposed in this PR:
- Reverts f062086c6b229ba5670d42ab23e94e5b6193cf15 (#929)
- Updates IDAES tag to 23.08.03 (@lbianchi-lbl)
- Changes to chemistry models for stability and to reflect upstream changes (https://github.com/IDAES/idaes-pse/pull/1197) (@lxhowl)
- Fixes initialization / testing issue with NF w/ bypass flowsheet and notebook (#1072) (@lxhowl)
- Remove call to `clone` from `watertap/unit_models/test/test_gac.py` for Pyomo 6.6.2 (@lxhowl)
- Better scaling in IX0D test (@lxhowl)
- Patch doctest for scaling factor warnings from NLv2 (@lxhowl)

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by bknueven** [comment]
Needs https://github.com/IDAES/idaes-pse/pull/1197

**Review by TimBartholomew — APPROVED** [review]
LGTM

**Comment by adam-a-a** [comment]
OK- I checked whether this PR would break anything in the BSM2 flowsheet by putting up a draft PR (#1108), and all checks pass! I'll review this and suspect it'll be ready to merge soon.

File: docs/how_to_guides/how_to_use_a_property_model.rst [hunk 1]
```diff
@@ -70,6 +70,12 @@ A portion of the displayed output is shown below.
 
 .. testoutput::
 
+   WARNING: model contains export suffix 'fs.state_block[0].scaling_factor' that
+   contains 4 component keys that are not exported as part of the NL file.
+   Skipping.
+   WARNING: model contains export suffix 'fs.state_block[0].scaling_factor' that
+   contains 4 component keys that are not exported as part of the NL file.
+   Skipping.
```
Comment (adam-a-a) [inline docs/how_to_guides/how_to_use_a_property_model.rst:78]: This looks like something we wouldn't want showing up in the how-to

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -304,4 +300,4 @@ jobs:
         pyomo build-extensions || python -c "from pyomo.contrib.pynumero.asl import AmplInterface; exit(0) if AmplInterface.available() else exit(1)"
     - name: Run pytest
       run: |
-        pytest --pyargs watertap -k 'not nf_dspmde.nf_ui'
+        pytest --pyargs watertap -k 'not (nf_dspmde.nf_ui or nf_dspmde.nf_with_bypass_ui)'
```
Comment (adam-a-a) [inline .github/workflows/checks.yml:303]: I suppose the alternative GUI flowsheet code was also causing checks to fail, hence it's addition here. DOes that mean the UI code for these NF flowsheets still has issues to address?

File: watertap/examples/chemistry/tests/test_pure_water_pH.py [hunk 3]
```diff
@@ -132,6 +132,7 @@ def base_units():
 def solver():
     s = get_solver()
     s.options["max_iter"] = 200
+    # s.options["halt_on_ampl_error"] = "yes"
```
Comment (adam-a-a) [inline watertap/examples/chemistry/tests/test_pure_water_pH.py]:
```suggestion
```

File: watertap/examples/chemistry/tests/test_pure_water_pH.py [hunk 4]
```diff
@@ -585,7 +619,8 @@ def test_initialize(self, model_initialize):
 
     @pytest.fixture
     def model_solve(self, model, solver):
-        results = solver.solve(model, tee=True)
+        model.write("bah.nl", io_options={"symbolic_solver_labels": True})
```
Comment (adam-a-a) [inline watertap/examples/chemistry/tests/test_pure_water_pH.py]: Why is this here?

File: watertap/examples/chemistry/tests/test_solids.py [hunk 5]
```diff
@@ -322,7 +324,7 @@ def run_case1(xA, xB, xAB=1e-25, scaling=True, rxn_config=None):
     )
     # End scaling if statement
 
-    solver.options["max_iter"] = 200
+    # solver.options["max_iter"] = 200
```
Comment (adam-a-a) [inline watertap/examples/chemistry/tests/test_solids.py]:
```suggestion
```

File: watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py [hunk 6]
```diff
@@ -67,28 +67,29 @@ def test_ideal_naocl_chlorination():
     ].value == pytest.approx(5.6407676871845223e-11, rel=1e-3)
 
 
+@pytest.mark.requires_idaes_solver
 @pytest.mark.component
 def test_ideal_naocl_chlorination_full_block():
     model = run_chlorination_block_example(fix_free_chlorine=True)
     assert model.fs.ideal_naocl_mixer_unit.dosing_rate.value == pytest.approx(
-        0.9504457542440085e-6, rel=1e-3
+        1.7296113311683092e-09, rel=1e-3
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py:75]: 3 order of magnitude change--not sure if this is correct due to inaccuracies in the original values used that were corrected or if this is an error itself. Without a deep dive into verifying replaced values, my guess is that this result is due to the former.

**Review by adam-a-a — COMMENTED** [review]
Just a couple comments, but overall, should be good to go.

File: .github/workflows/checks.yml (hunk 2 above)
Comment (lbianchi-lbl) [reply .github/workflows/checks.yml:303]: IIRC this is due to a limitation of the `watertap.ui.fsapi` layer for models that require the "good" (`ma27`-enabled) Ipopt to solve. If the model can be initialized without failures, specifying `requires_idaes_solver=True` kwarg is enough to skip the corresponding test without failures, but this doesn't work when the solve fails  during initialization. Therefore, we have to exclude the entire test module using the `-k not ...` pytest flag.

File: docs/how_to_guides/how_to_use_a_property_model.rst (hunk 1 above)
Comment (bknueven) [reply docs/how_to_guides/how_to_use_a_property_model.rst:78]: 1. It's just in the test -- so the reader won't see it.
2. I talked with @andrewlee94 about this a few weeks ago and there is not a workaround nor is one planned (IDAES and Pyomo are implementing a new model scaling framework)

File: watertap/examples/chemistry/tests/test_pure_water_pH.py [hunk 7]
```diff
@@ -585,7 +620,8 @@ def test_initialize(self, model_initialize):
...
```
Comment (bknueven) [inline watertap/examples/chemistry/tests/test_pure_water_pH.py]:
```suggestion
```
Debugging

File: .github/workflows/checks.yml (hunk 2 above)
Comment (bknueven) [reply .github/workflows/checks.yml:303]: Note -- this only applies to the macOS tests

File: watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py (hunk 6 above)
Comment (bknueven) [reply watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py:75]: `dh_rxn_ref` was *also* corrected by three orders of magnitude above.

There's also other changes to how the solubility equilibrium constraint is implemented in IDAES https://github.com/IDAES/idaes-pse/pull/1197

Comment (adam-a-a) [reply watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py:75]: OK, so due to the first reason I mentioned. Thanks for confirming.

**Review by adam-a-a — APPROVED** [review]
LGTM
